src/librustc_mir/transform/deaggregator.rs

```diff
@@ -21,116 +21,103 @@ impl MirPass for Deaggregator {
                           tcx: TyCtxt<'a, 'tcx, 'tcx>,
                           source: MirSource,
                           mir: &mut Mir<'tcx>) {
-        let node_path = tcx.item_path_str(source.def_id);
-        debug!("running on: {:?}", node_path);
-        // we only run when mir_opt_level > 2
-        if tcx.sess.opts.debugging_opts.mir_opt_level <= 2 {
-            return;
-        }
-
         // Don't run on constant MIR, because trans might not be able to
         // evaluate the modified MIR.
         // FIXME(eddyb) Remove check after miri is merged.
         let id = tcx.hir.as_local_node_id(source.def_id).unwrap();
         match (tcx.hir.body_owner_kind(id), source.promoted) {
```

Comment:
Nit: I find this match a bit hard to read. Maybe something like this?

```rust
let is_constant = source.promoted.is_some() || match tcx.hir.body_owner_kind(id) {
    hir::BodyOwnerKind::Const | hir::BodyOwnerKind::Static(_) => true,
    hir::BodyOwnerKind::Fn => tcx.is_const_fn(source.def_id),
};
if is_constant {
    return;
}
```

Reply:
It's copied from elsewhere, this pattern is used in several passes - and this will hopefully go away ~soon anyway.